### PR TITLE
feat: added initial Github CI workflow

### PR DESCRIPTION
**Pull request by airween**
Added .github/workflow/ci.yml and necessary files with a complex test workflow for each PR.

A brief summary about how does it work:

* set up dependencies (required packages)
* install latest ModSecurity3 release
* get ModSecurity-nginx source
* get latest Nginx soure
* build Nginx with ModSecurity module
* start Nginx with ModSecurity
* there are two virtual hosts
* there is only one rule
* run a valid request (expected code is 200) and an attack (expected code is 403) against both vhosts

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,89 @@
+name: Quality Assurance
+
+on:
+  push:
+  pull_request:
```
Comment (theseion) [inline .github/workflows/test.yml:5]:
```suggestion
  pull_request:
  
  # Default: none
  permissions: {}
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+jobs:
```
Comment (theseion) [inline .github/workflows/test.yml:7]:
```suggestion
jobs:
  permissions:
    contents: read
```

**Review by theseion — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  build-linux:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-22.04]
+        compiler: [gcc, clang]
+    env:
+        CC: "/usr/bin/${{ matrix.compiler }}"
+        CXX: "${{ matrix.compiler == 'gcc' && '/usr/bin/g++' || '/usr/bin/clang++' }}"
```
[the remark this answers is not in the document]
Comment (airween) [reply .github/workflows/test.yml]: Done.

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+      - name: Get Nginx source
+        uses: actions/checkout@v4
+        with:
+          repository: nginx/nginx
+          path: nginx
+          fetch-depth: 1
+      - name: Build nginx with ModSecurity-nginx module
+        working-directory: nginx
+        run: |
+          ./auto/configure --with-ld-opt="-Wl,-rpath,/usr/local/lib" --without-pcre2 --add-module=/home/runner/work/ModSecurity-nginx/ModSecurity-nginx/ModSecurity-nginx
```
[the remark this answers is not in the document]
Comment (airween) [reply .github/workflows/test.yml]: Changed to relative path, but this form didn't work.

File: .github/workflows/test.yml (hunk 1 above)
Comment (airween) [reply .github/workflows/test.yml:5]: Put the `permissions` block without indentation.

File: .github/workflows/test.yml (hunk 2 above)
Comment (airween) [reply .github/workflows/test.yml:10]: Put the `permission` block under the `build-linux` block.

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+          make
+          make modules
+          sudo make install
+      - name: Start Nginx
+        run: |
+          sudo /usr/local/nginx/sbin/nginx -c /home/runner/work/ModSecurity-nginx/ModSecurity-nginx/ModSecurity-nginx/.github/nginx/nginx.conf
```
[the remark this answers is not in the document]
Comment (airween) [reply .github/workflows/test.yml:57]: This does not work, I must have to keep the absolute path for the config. But I think we are fine with that.

**Review by theseion — APPROVED** [review]

**Comment by airween** [comment]
Many-many thanks for @theseion for your big help!

**Pull request merged**